pkg/build/controller/build/build_controller.go

```diff
@@ -966,7 +966,18 @@ func (bc *BuildController) resolveImageStreamTag(ref *corev1.ObjectReference, li
 		}
 		return nil, fmt.Errorf("the referenced image stream %s/%s could not be found: %v", namespace, name, err)
 	}
-	if newRef, ok := imageutil.ResolveLatestTaggedImage(stream, tag); ok {
+	newRef := ""
+	ok = false
+	if len(stream.Status.DockerImageRepository) > 0 {
+		newRef, ok = imageutil.ResolveLatestTaggedImage(stream, tag)
+	} else {
+		streamCopy := stream.DeepCopy()
+		// in case the api server has not yet picked up the internal registry hostname from the cluster wide
+		// OCM config, we use our copy here to facilitate leveraging pull through with local tag reference policy
+		streamCopy.Status.DockerImageRepository = bc.internalRegistryHostname
+		newRef, ok = imageutil.ResolveLatestTaggedImage(streamCopy, tag)
+	}
```

Comment:
so is this the only place where we ever resolve an imagestreamtag for a buildconfig? 

i know we potentially allow imagestreamtag references in various places in buildconfigs(e.g. image inputs in addition to builder images), hopefully they all go through here but have you sanity checked that?

Reply:
`image_trigger_controller.go` makes a call to `imageutil.ResolveLatestTaggedImage` , and some of the builconfig image change trigger methods make calls to methods that call it.

But it seemed in those cases, the flow are driven off of imagestream events, so in theory we have an updated imagestream and should in theory have that image registry hostname set as part of getting the event.

Concur ?

Otherwise, all the build_controller flows, as part of creating the build pod, go through this method.

Reply:
yeah i think that sounds ok.

Reply:
cool thanks